### PR TITLE
EIA923 Early Release!

### PR DESCRIPTION
**Pull request by cmgosnell**
- [x] metadata
- [x] runs through extract
- [x] runs through transform
- [x] runs through the etl w/ just eia (w/o PK constraints)
- [x] add unmapped plant/utility ids (if any)
- [x] runs through the full etl 
- [x] passes data validation tests

### CODE REVIEW

**Comment by zaneselvans** [comment]
If we're going to integrate the Early Release data into the DB more generally (which seems great, if it's easy to integrate) would it make sense to keep the ER / not-ER flag column, and add it to the DB?  Would it be clear which records come from the ER data, or does that get too muddy in the normalization process? Could we just add a flag across the board for the monthly and annual data from the EIA-923 and EIA-860 ER year?

**Comment by cmgosnell** [comment]
@zaneselvans  this is basically what i am suggestions in #1799

**Comment by cmgosnell** [comment]
Well! the data runs through the eia-only etl just fine rn.

I ran the `./devtools/ferc1-eia-glue/find_unmapped_plants_utils.py` script and the only unmapped anything was some plants that had NO INFO. I went back and checked... and these plants are very very inconsistent in the early release data across various tabs. So the harvesting process is doing a good job here nulling out  this data. I've temporarily added these plant id's into the pudl id mapping spreadsheet... but we should go back to these once we add the 860 early release.

image

File: src/pudl/transform/eia923.py [hunk 1]
```diff
@@ -636,6 +636,7 @@ def generation_fuel(eia923_dfs, eia923_transformed_dfs):
         "total_fuel_consumption_mmbtu",
         "elec_fuel_consumption_mmbtu",
         "net_generation_megawatthours",
+        "early_release",
```
Comment (cmgosnell) [inline src/pudl/transform/eia923.py:639]: I've added these new `early_release` columns in here as columns to drop *for now*. I'd like to put this aside until we get somewhere with #1799 (already have an implement issue #1796

**Comment by zaneselvans** [comment]
Does it make sense to merge this PR without the EIA 860 Early Release data? We have the generators table from 2021 thanks to the eia860m, but we won't have any of the other 2021 data -- plants, utilities, new BGAs, etc.  Do you think the EIA-860ER will be similarly straightforward?

**Comment by cmgosnell** [comment]
These are the exact questions I asked our team. I am already working on the eia860 er and i *hope* it'll be similarly straightforward. I don't think there is a real harm in integrating the new metadata and slight code updates. I would suggest not actually changing the datasource working partitions until they are both integrated, but merging this pr.

**Comment by zaneselvans** [comment]
So use the new data archives / DOIs which include all the old data (however it's changed...) plus the 2021 eia923er data, but not actually allow that data to be processed until we get the eia860er integrated as well.

The new expected numbers of rows only apply if the 2021 eia923er data is processed though right?

**Comment by zaneselvans** [comment]
(I merged in `dev` to fix the dependency issue that was causing the push to fail CI)

**Comment by cmgosnell** [comment]
consolidated into #1834

**Pull request closed without merging**